config.c

```diff
@@ -59,11 +69,116 @@ void init_default_style(struct mako_style *style) {
 	memset(&style->spec, true, sizeof(struct mako_style_spec));
 }
 
+void init_empty_style(struct mako_style *style) {
+	memset(style, 0, sizeof(struct mako_style));
+}
+
 void finish_style(struct mako_style *style) {
 	free(style->font);
 	free(style->format);
 }
 
+// Update `target` with the values specified in `style`. If a failure occurs,
+// `target` will remain unchanged.
+bool apply_style(struct mako_style *style, struct mako_style *target) {
```

Comment:
Generally in C we prefer to have the struct on which the function operates always as the first argument. It seems the comment "`target` will remain unchanged" is outdated.

Also to make it more obvious that an argument isn't mutated, you can make it `const`.